pkg/monitoring/rules/recordingrules/virt.go

```diff
@@ -53,7 +53,7 @@ func virtRecordingRules(namespace string) []operatorrules.RecordingRule {
 			},
 			MetricType: operatormetrics.GaugeType,
 			Expr: intstr.FromString(
-				fmt.Sprintf("sum(kubevirt_virt_controller_ready_status{namespace='%s'}) or vector(0)", namespace),
+				fmt.Sprintf("count(kube_pod_status_ready{pod=~'virt-controller-.*', namespace='%s', condition='true'} + on(pod, namespace) kubevirt_virt_controller_ready_status{namespace='%s'} != 2) or vector(0)", namespace, namespace),
```

Comment:
this improves upon the current readiness check of the pod, instead of only relying on it reporting its status, we should also include `kube_pod_status_ready` which is based on the readiness probe of the deployments